gavel_ci/ui/jobserv.py

```diff
@@ -148,6 +160,24 @@ def run(proj, build, run):
     return render_template('run.html', project=proj, build=build, run=run)
 
 
+@blueprint.route('projects/<project:proj>/builds/<int:build>/<run>/'
+                 'console')
+def console(proj, build, run):
+    return render_template('console.html', proj=proj, build=build, run=run)
+
+
+@blueprint.route('projects/<project:proj>/builds/<int:build>/<run>/'
+                 'console/tail')
+def console_tail(proj, build, run):
+    response = _raw_get(
+        f'/projects/{proj}/builds/{build}/runs/{run}/console.log',
+        headers={
+            'X-OFFSET': request.headers.get('X-OFFSET', '0'),
+            'Range': request.headers.get('Range', ''),
```

Comment:
Sorry for the delay, I forgot this project is not under the company umbrella and I must be getting notifications in a different email. 

Back to your question, while JobServ is handling `console.log`, API requests do not respond to `Range` requests.  Whatever is serving `console.log` *after* a build is finished *does* support Range requests.   

I use this fact, so that when a user comes to look at the logs after a build is finished, it only loads the last 2kb of the log to show the result of the build.  The link still allows them to view the entire raw log.

Reply:
we'll need to move this to foundriesio. but that's another day.